packages/eslint-plugin/src/util/getTextWithParentheses.ts

```diff
@@ -0,0 +1,35 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import type { SourceCode } from '@typescript-eslint/utils/ts-eslint';
+
+import {
+  isClosingParenToken,
+  isOpeningParenToken,
+  isParenthesized,
+  nullThrows,
+  NullThrowsReasons,
+} from '.';
```

Comment:
We generally try not to import `from '.'`, as that generates a cycle:

```suggestion
} from '../util';
```

Reply:
I can do that, though that's still a cycle; those are the same imports, they both point to `index.ts`.

Reply:
Oh, maybe we need to flatten these out...